nextcord/user.py

```diff
@@ -85,6 +86,74 @@ def asset(self) -> Asset:
         return Asset._from_avatar_decoration(self.user._state, self._asset)
 
 
+class PrimaryGuild:
+    """Represents a primary guild for a user.
+
+    .. versionadded:: 3.2
+
+    Attributes
+    ----------
+    identity_guild_id: Optional[:class:`int`]
+        The ID of the primary guild.
+    identity_enabled: Optional[:class:`bool`]
+        Whether displaying the primary guild is enabled.
+
+        This is ``None`` if the user has not accepted a change to the tag details.
+    tag: Optional[:class:`str`]
+        The tag of the primary guild.
+    badge: Optional[:class:`str`]
+        The badge of the primary guild.
+    """
+
+    __slots__ = (
+        "_state",
+        "identity_guild_id",
+        "identity_enabled",
+        "tag",
+        "_badge",
+    )
+
+    def __init__(self, *, data: UserPrimaryGuildPayload, state: ConnectionState) -> None:
+        self._state: ConnectionState = state
+        self.identity_guild_id: Optional[int] = get_as_snowflake(data, "identity_guild_id")
+        self.identity_enabled: Optional[bool] = data.get("identity_enabled")
+        self.tag: Optional[str] = data.get("tag")
+        self._badge: Optional[str] = data.get("badge")
+
+    @classmethod
+    def _empty(cls, state: ConnectionState) -> Self:
+        """Creates an empty :class:`PrimaryGuild`.
+
+        This is a better alternative to returning ``None`` when the primary guild is not set,
+        as all of these fields are nullable anyway, allows for easier checking of the primary guild.
+        """
+        return cls(
+            data={
+                "identity_guild_id": None,
+                "identity_enabled": None,
```

Comment:
`identity_enabled` being `None` is generally for when the primary guild has been edited but the user has not reaffirmed the guild identity after a change. 
```suggestion
                "identity_enabled": False,
```